src/radio/radio-controller.ts

```diff
@@ -165,4 +166,101 @@ export class RadioController {
       }),
     ]);
   }
+
+  /**
+   * Sends an HTTP GET request using the provided parameters.
+   * @param host  The address to send the HTTP GET request to.
+   * @param port  The port to send the HTTP GET request to.
+   * @param path  The path to send the HTTP GET request to.
+   * @param timeout  How long to wait before rejecting with a timeout.
+   */
+  public async sendHttpGet(
+    host: string,
+    port: number,
+    path: string,
+    timeout = RADIO_TIMEOUT
+  ): Promise<smarthome.DataFlow.HttpResponse> {
+    const options = {
+      host,
+      port,
+      path,
+    };
+    const httpResponse = new Promise<smarthome.DataFlow.HttpResponse>(
+      resolve => {
+        http.get(options, res => {
+          const {statusCode} = res;
+          if (statusCode !== 200) {
+            throw new Error(`Request Failed.\nStatus Code: ${statusCode}`);
+          }
+          res.setEncoding('utf8');
+          res.on('data', chunk => {
+            const httpResponse = new HttpResponse(chunk, statusCode);
```

Comment:
You may want to consider using a string buffer, as there's no guarantee that there will only be one chunk. The `end` event provides more guidance on when the response actually ends.

Reply:
Fixed this. I used a string array and `.join('')` when resolving.